cpp/memilio/compartments/simulation_base.h

```diff
@@ -78,11 +100,11 @@ class SimulationBase
      */
     Core& get_integrator()
     {
-        return *m_integratorCore;
+        return m_integrator.get_integrator();
```

Comment:
Should we maybe rename the new function to sth like get_core or get_integrator_core? Or maybe we could change the name of m_integrator? It is okay as is, but I do think `integrator.get_integrator()` looks a little weird.